src/System.Reflection.MetadataLoadContext/src/System/Reflection/MetadataLoadContext.Apis.cs

```diff
@@ -0,0 +1,298 @@
+    //
+    //       m1 == m2
+    //       m1.Equals(m2)
+    //
+    //   but not
+    //
+    //       object.ReferenceEquals(m1, m2)   // WRONG
+    //       (object)m1 == (object)m2         // WRONG
+    //
+    //   Note that the following descriptions are not literal descriptions of how Equals() is implemented. The MetadataLoadContext
+    //   reserves the right to implement Equals() as "object.ReferenceEquals()" and intern the associated objects in such
+    //   a way that Equals() works "as if" it were comparing those things.
+    //
+    // - Each MetadataLoadContext permits only one Assembly instance per assembly identity so equality of assemblies is the same as the
+    //   equality of their assembly identity.
+    // 
+    // - Modules are compared by comparing their containing assemblies and their row indices in the assembly's manifest file table.
+    // 
+    // - Defined types are compared by comparing their containing modules and their row indices in the module's TypeDefinition table.
+    // 
+    // - Constructed types (arrays, byrefs, pointers, generic instances) are compared by comparing all of their component types.
+    // 
+    // - Generic parameter types are compared by comparing their containing Modules and their row indices in the module's GenericParameter table.
+    // 
+    // - Constructors, methods, fields, events and properties are compared by comparing their declaring types, their row indices in their respective
+    //   token tables and their ReflectedType property.
+    // 
+    // - Parameters are compared by comparing their declaring member and their position index.
+    //
+    // Multithreading:
+    //    
+    //   The MetadataLoadContext and the reflection objects it hands out are all multithread-safe and logically immutable,
+    //   except that no Loads or inspections of reflection objects can be done during or after disposing the owning MetadataLoadContext.
+    //
+    // Support for NetCore Reflection apis:
+    //
+    //   .NETCore added a number of apis (IsSZArray, IsVariableBoundArray, IsTypeDefinition, IsGenericTypeParameter, IsGenericMethodParameter,
+    //      HasSameMetadataDefinitionAs, to name a few.) to the Reflection surface area.
+    //
+    //   The Reflection objects dispensed by MetadataLoadContexts support all the new apis *provided* that you are using the netcore build of System.Reflection.MetadataLoadContext.dll.
+    //
+    //   If you are using the netstandard build of System.Reflection.MetadataLoadContext.dll, the NetCore-specific apis are not supported. Attempting to invoke
+    //   them will generate a NotImplementedException or NullReferenceException (unfortunately, we can't improve the exceptions thrown because
+    //   they are being thrown by code this library doesn't control.) Because of this, it is recommended that apps built against NetCore use the NetCore build of
```

Comment:
Apps built against NetCore 3.0 will automatically use the NetCore build of this library. They do not really have a choice:  They would need to go out of your way to convince NuGet to not give them NetCore build of this library when built against .NetCore 3.0.